code-challenges.test.js

```diff
@@ -22,7 +24,38 @@ const fibonacciLength2 = 10
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+//Jest Testing: 
+// describe("fibSequence", () => {
+//   it("Returns an array that is the length of the given input. It adds the numbers inside the array, but in Fibonacci sequence", () => {
+//     expect(fibSequence(fibonacciLength1)).toEqual([1, 1, 2, 3, 5, 8])
+//     expect(fibSequence(fibonacciLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+//   })
+// })
+
+//Psuedocode
+//Function Name: fibSequence 
+//Input: Number
+//Output: An array with the length of the input number. 
+//Process: Turn given input into the length of the array. Once we have the length we will need to have 1 as the first value, 1 as the second value to add to 2. 2 as the third value to add to three and so on. An if statement to allow only numbers bigger than 2 to be added as the length of the array. If statement to see if number inputed is going to be greater than two, this will be the length of the array we are making. Make an array using Array(). A for loop to add numbers into array. Add logic inside for loop to make Fibonacci sequence work. 
+
+// const fibSequence = (num) => {
+//   if(num > 2){
+//     let arrayLength = Array(num)
+//     let startFibArray = [1,1]
+//     //Stuck on trying to make the logic of the for loop work. Make the fibonacci logic.
+//     for (let i = 2; i < arrayLength.length; i++) {
+//       // [1 + 1 + 2 + 3 + 5 + 8]
+//       startFibArray[i] = startFibArray[i-2]+startFibArray[i-1]
+//      }
+//     //Was able to add numbers into empty array by moving return out of the for loop. Kept adding it into for loop which would stop the loop after the one time.
+//     return startFibArray
+//   }
+// }
+// console.log(fibSequence(fibonacciLength1))
+// console.log(fibSequence(fibonacciLength2))
+//Cannot figure out the logic for the life of me.
+//Had to get some help to figure out the logic inside the for loop. 
```

Comment:
The logic here starts with the `startFibArray` variable, which is set to `[1, 1]`. Then, you are starting an iteration with a for loop, starting on index 2 and ending when it reaches the index of whatever number was passed in as an argument. Each iteration assigns the current index value to the previous 2 values added together (`startFibArray[i] = startFibArray[i-2]+startFibArray[i-1]`). This goes on until the array reaches the desired length, because the for loop is stopping iteration depending on the number that was passed as an argument.

Some further elaboration: on the first iteration, `startFibArray` is still `[1, 1]`. Since the for loop is starting at index 2, that will be the 3rd number in the array. For the first iteration, the logic in the for loop is saying "the third number is the first number plus the second number". For the second iteration, the `startFibArray` variable is now `[1, 1, 2]`, and the logic is saying "the fourth number is the second number plus the third number". So on and so forth. Hope that helped!